torch/distributed/pipelining/stage.py

```diff
@@ -478,6 +478,7 @@ def backward_maybe_with_nosync(self, backward_type, bwd_kwargs: Dict):
             last_backward = self._seen_bwd_chunks == self.chunks - 1  # type: ignore[operator]
         else:
             # For backwards are split into weight and input, we will see twice as many bwd_chunks
+            # -1 because we skip the first bwd_chunk backward
```

Comment:
For another PR.. but in case you didn't see my comment on my own PR for merge bw, this logic will have to be rewritten since any stage may have some mix of I, W and B operations so we can't expect to do it by counting and expecting round numbers.

Reply:
Got it, we can definitely rewrite this logic!